pylint/lint/pylinter.py

```diff
@@ -537,7 +537,7 @@ def make_options() -> Tuple[Tuple[str, OptionDict], ...]:
     def __init__(
         self,
         options=(),
-        reporter=None,
+        reporter: Union[reporters.BaseReporter, reporters.MultiReporter, None] = None,
```

Comment:
I wanted to do this in a separate PR but some of the issues fixed here didn't show if I didn't add typing to at least one of these parameters.